src/ConcreteOperations/volume.jl

```diff
@@ -23,16 +23,27 @@ function volume(H::AbstractHyperrectangle{N}) where {N<:Real}
     return vol
 end
 
-function volume(B::BallInf{N}) where {N<:Real}
+@inline function _vol_prod(H::AbstractHyperrectangle{N}, n) where {N<:Real}
+    vol = one(N)
+    for i in 1:n
+        vol *= 2 * radius_hyperrectangle(H, i)
+    end
+    return vol
+end
+
+function volume(B::BallInf{N}) where {N<:AbstractFloat}
     n = dim(B)
     if n < 50
-        vol = one(N)
-        for i in 1:n
-            vol *= 2 * radius_hyperrectangle(B, i)
-        end
+        vol = _vol_prod(B, n)
     else
         r = B.radius
         vol = exp(n*log(2r))
     end
     return vol
 end
+
+function volume(B::BallInf{N}) where {N<:Real}
+    n = dim(B)
+    vol = _vol_prod(B, n)
```

Comment:
Since you optimized for `BallInf`: multiplying `2 * r` `n` times has `2n` multiplications, but you can do with only `n+1`:
```julia
    diameter = 2 * B.radius
    return diameter^dim(B)
```

Reply:
I tried it but it didnt show to be faster.. maybe because of the `^` operation?
Feel free to open a follow-up issue.

Reply:
Then at least this one should be faster.
```julia
@inline function _vol_prod(B::BallInf{N}, n) where {N<:Real}
    vol = one(N)
    diameter = 2 * radius_hyperrectangle(H, i)
    for i in 1:n
        vol *= diameter
    end
    return vol
end
```

Reply:
I've added it, although in the tests it doesn't come up as faster than the current one. I wouldn't be surprised if the compiler already does some tricks to optimize the loop.

Reply:
Okay, thanks for trying.